app/models/accounting/quickbooks/full_fetcher.rb

```diff
@@ -10,20 +10,56 @@ def initialize(qb_connection = Division.root.qb_connection)
       end
 
       def fetch_all
-        divisions_accounts = Division.clear_all_accounts
+        divisions_accounts = clear_all_accounts!
 
         ::Accounting::Transaction.destroy_all
         ::Accounting::Account.destroy_all
 
         started_fetch_at = Time.zone.now
 
-        ::Accounting::Quickbooks::AccountFetcher.new.fetch
-        ::Accounting::Quickbooks::TransactionFetcher.new.fetch
+        ::Accounting::Quickbooks::AccountFetcher.new(@qb_connection).fetch
+        ::Accounting::Quickbooks::TransactionFetcher.new(@qb_connection).fetch
 
         qb_connection.update_attribute(:last_updated_at, started_fetch_at)
 
-        Division.restore_all_accounts(divisions_accounts)
+        restore_all_accounts!(divisions_accounts)
       end
+
+      private
+
+      # Set all divisions' accounts to nil and return a hash of the QB ids of the removed accounts
+      # by division
+      def clear_all_accounts!
+        divisions_accounts = {}
+        Division.all.each do |d|
+          accounts_qb_ids = {
+            principal_qb_id: d.principal_account&.qb_id,
+            interest_receivable_qb_id: d.interest_receivable_account&.qb_id,
+            interest_income_qb_id: d.interest_income_account&.qb_id,
+          }
+          d.update(
+            principal_account_id: nil,
+            interest_receivable_account_id: nil,
+            interest_income_account_id: nil,
```

Comment:
I'm wondering why we're not deleting the account objects instead of just nullifying the foreign keys... Is there any reason to keep them around? @linesarefuzzy thoughts?

Reply:
They’re all about to get deleted anyway. Why delete them in two places?

Reply:
Ok I didn't realize that, we cool.